test/system/setup_test.go

```diff
@@ -44,8 +44,27 @@ func RunTestSuites(t *testing.T) {
 	suite.Run(t, new(DslTestSuite))
 }
 
-func DumpAndRun(compress bool) {
-	ctx := context.Background()
+func InitSetupTest(ctx context.Context) *providers.ProvidersFactoryConfig {
+	err := cmd.InitializeKubehoundConfig(ctx, KubeHoundThroughDumpConfigPath, false, false)
+	if err != nil {
+		log.I.Fatalf(err.Error())
+	}
+
+	khCfg, err := cmd.GetConfig()
+	if err != nil {
+		log.I.Fatal(err.Error())
+	}
+
+	// Initialisation of the p needed for the ingestion and the graph creation
+	p, err := providers.NewProvidersFactoryConfig(ctx, khCfg)
+	if err != nil {
+		log.I.Fatalf("factory config creation: %v", err)
+	}
+
+	return p
+}
+
+func DumpAndRun(ctx context.Context, compress bool, providers *providers.ProvidersFactoryConfig) {
```

Comment:
You should probably rename the `providers` argument, it is overriding the `providers` package and make the code a bit harder to understand

Reply:
changed